### ISSUE TITLE
[Bug] [CMake] libtvm_runtime and cpp rpc won't build with NDK 23

### ISSUE THREAD

**Issue by lhez** [apache/tvm#9173]
NDK 23 seems to make some major changes from previous releases - it removes all gcc related components and `ld.gold`. This causes two issues when building with NDK 23.

First, `libgcc` is added in the root [`CMakeLists.txt`](https://github.com/apache/tvm/blob/main/CMakeLists.txt#L228),

```
if(BUILD_FOR_ANDROID)
  # EmuTLS on Android is in libgcc. Without it linked in, libtvm_runtime.so
  # won't load on Android due to missing __emutls_XXX symbols.
  list(APPEND TVM_RUNTIME_LINKER_LIBS "gcc")
endif()
```
This will result in following linker error because `libgcc` has been removed from NDK 23 (instead `libclang_rt.builtins-aarch64-android.a` is used)

```
ld: error: unable to find library -lgcc
```

This can be reproduced with the following commands,

```
cmake -DCMAKE_TOOLCHAIN_FILE=/ndk/23.0.7599858/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DANDROID_PLATFORM=28 -DANDROID_STL=c++_static -DUSE_CPP_RPC=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -G Ninja ..
ninja tvm_rpc
```

Linker will fail with `libtvm_runtime.so`. This can be fixed by disabling `libgcc` for NDK is 23. I am not sure about the TLS functions - but `tvm_rpc` runs fine with NDK 23 for me when `libgcc` is removed.

Second, when OpenCL is enabled, `tvm_rpc` (the cpp rpc) will use `ld.gold` as the linker. This is specified in `tvm_rpc`'s [`CMakeLists.txt`](https://github.com/apache/tvm/blob/main/apps/cpp_rpc/CMakeLists.txt#L35),

```
if(USE_OPENCL)
  if (ANDROID_ABI)
    set_property(TARGET tvm_rpc PROPERTY LINK_FLAGS -fuse-ld=gold)
  endif()
endif()
```

Since `ld.gold` is removed from NDK 23, Clang seems to call host `ld.gold`, resulting in failure

```
/usr/bin/ld.gold: --no-rosegment: unknown option
/usr/bin/ld.gold: use the --help option for usage information
```

This can be reproduced by adding OpenCL,

```
cmake -DCMAKE_TOOLCHAIN_FILE=/ndk/23.0.7599858/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -D
ANDROID_PLATFORM=28 -DANDROID_STL=c++_static -DUSE_CPP_RPC=ON -DUSE_OPENCL=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -G Ninja ..
ninja tvm_rpc
```

Similarly, this can be fixed by disabling `ld.gold` for NDK 23.

**Comment by echuraev** [comment]
I think this bug was fixed in this #13118 and other PRs which were mentioned in https://github.com/apache/tvm/pull/13118.

@lhez, @areusch, @hpanda-naut, I suppose to close this issue.

**Comment by lhez** [comment]
Yes, this has been fixed in #13118. I should have closed this issue when #13118 was merged.
